feat: add docs about limiting API access by IP (#1924)
buildkite/docs: pages/apis/managing_api_tokens.md

>📘
> Removing access from a token does not delete the token, however token owner cannot re-add your organization to their token's scope.

> ## Limiting API Access by IP address

Reviewer: Nitpick: Add a blank line to separate elements like headings and paragraphs.

```suggestion
## Limiting API Access by IP address

```

>🚧 Beta Feature

> > This feature is not yet available to all organizations.

Reviewer: I'll follow up about this part in a DM

Reviewer 2: Hey @jfhinchcliffe (@mbelton-buildkite)

Yeah, I was just trying to be consistent across the board with the communication style.

My concern primarily stems from the GraphQL implementation. 

Given that this mutation becomes public knowledge and discoverable via the GraphQL explorer or introspection queries. 

If it _looks_ to be functional, but isn't functional due to the feature flag then I think we have a problem. Especially when we're dealing with security related features like an Allowlist. 

However, I'm happy to defer to @mbelton-buildkite for best practices here.

Author: changed to 'Private Beta' and updated screenshot above 😁

Author: Removed 'Beta' wording.

> If you'd like to limit access to your organization by IP address, you can create an allow list of IP addresses on the [organization's member permissions page](https://buildkite.com/organizations/~/member-permissions).

Reviewer: - "Allowlist" is one word.
- TIL that you can format links like that to autocomplete the org in the URL. 🔥
- I've tried a different way of describing the location that I think is clearer, but let me know what you think.

```suggestion
If you'd like to limit access to your organization by IP address, you can create an allowlist of IP addresses in the [organization's permission settings](https://buildkite.com/organizations/~/member-permissions).
```

Reviewer 2: > "Allowlist" is one word.

TIL... i've been writing it as Allow list

> The allow list can also be managed using the [GraphQL API](/docs/apis/graphql-api) with the `organizationAPIIPAllowListUpdate` mutation.

Reviewer: Nitpicks:

- allow list -> allowlist
- Passive voice -> active
- Link to the specific mutation (I assume that's the one you were waiting for)

```suggestion
You can also manage the allowlist with the [`organizationAPIIPAllowListUpdate`](/docs/apis/graphql/schemas/mutation/organizationapiipallowlistupdate) mutation in the GraphQL API.
```

Reviewer: Is it too late to rename the mutation? I think it should be `organizationApiIpAllowlistUpdate `

Author: Nope! PR here: https://github.com/buildkite/buildkite/pull/11543

Author: Above PR merged. Updated to `organizationApiIpAllowlistUpdate` here.
